File: oo/oo.py
class Language:
    _all = []

    def __init__(self, name, eng_name):
        self.name = name
        self.in_english = eng_name
        self.words = {}
        self._save()

    def _save(self):
        self._all.append(self)
    
    def add_word(self, english, translation):
        self.words[english] = translation
    
    def _translate(self, word):
        return self.words[word]

    def how_do_you_say(self, word):
        print(f"In {self.name}, '{word}' is '{self._translate(word)}'.")

    @classmethod
    def print_all(cls):
        print("We have the following languages:")
        for lang in cls._all:
            print(lang.name)


spanish = Language("Español", "Spanish")
german = Language("Deutsch", "German")

spanish.add_word("Hello", "Hola")
spanish.how_do_you_say("Hello")


Language.print_all()

File: oo/oo.rb
class Language
    attr_accessor :name, :in_english
    @@all = []

    def initialize(name, eng_name)
        @name = name
        @in_english = eng_name
        @words = {}
        save
    end

    def save
        @@all << self
    end

    def add_word(english, translation)
        @words[english] = translation
    end

    def how_do_you_say(word)
        puts "In #{name}, '#{word}' is '#{translate(word)}'."
    end

    def self.print_all
        puts "We have the following languages:"
        @@all.each{|lang| puts lang.name}
    end

    private
    def translate(word)
        @words[word]
    end
end

spanish = Language.new("Español", "Spanish")
german = Language.new("Deutsch", "German")

spanish.add_word("Hello", "Hola")
spanish.how_do_you_say("Hello")

Language.print_all